docs/pages/cairo/enum.md

```diff
@@ -1,21 +1,76 @@
 ## Enum
 
-Enums are very useful in game design, as they simplify the creation of clean, complex logic.
+## What is Enum
+Enums, short for "enumerations," are a way to define a custom data type that consists of a fixed set of named values, called variants. Enums are useful for representing a collection of related values where each value is distinct and has a specific meaning. Enums are particularly useful in game development for representing game states, player actions, or any other set of related constants that a game might need to track.
 
-You can define an enum as follows:
+In this example, we've defined an enum called PlayerCharacter with four variants: `Godzilla`, `Dragon`, `Fox`, and `Rhyno`. The naming convention is to use `PascalCase` for enum variants. Each variant represents a distinct value of the `PlayerCharacter` type. In this particular example, variants don't have any associated value. 
+Now let's imagine that our variants have associated values, We can define a new PlayerCharacter enum:
 
 ```rust
+# [derive(Serde, Copy, Drop, Introspect, PartialEq, Print)]
+enum PlayerCharacter {
+    Godzilla: u128,
+    Dragon:   u128,
+    Fox:      u128,
+    Rhyno:    u128
+}
+
+```
+However, in Dojo, all enum variants must share the same type. This limitation poses a challenge when we want to associate different data types with each variant, as is often the case in game development where characters, items, or enemies may have unique attributes.
+To work around this limitation, we can use a struct to encapsulate the attributes of each character type. This allows us to maintain the unique characteristics of each character while adhering to the constraints.
+
+```rust
+# [derive(Serde, Copy, Drop, Introspect, PartialEq, Print)]
+enum PlayerCharacter {
+    Godzilla(CharacterAttributes),
+    Fox(CharacterAttributes),
+    Rhyno(CharacterAttributes),
+}
+
+# [derive(Serde, Copy, Drop, Introspect, PartialEq, Print)]
+struct CharacterAttributes {
+    is_godzilla: bool,
+    fox_value: Option<u32>, // Assuming felt252 is represented as u32 for simplicity
+    rhyno_values: Option<(u128, u128)>,
+}
+
+let godzilla = PlayerCharacter::Godzilla(CharacterAttributes {
+    is_godzilla: true,
+    fox_value: None,
+    rhyno_values: None,
+})
+
+let fox = PlayerCharacter::Fox(CharacterAttributes {
+    is_godzilla: false,
+    fox_value: Some(90), // Assuming felt252 is represented as u32 for simplicity
+    rhyno_values: None,
+});
+```
+For the `Godzilla` variant, since it doesn't require any additional data, you would instantiate it with a `CharacterAttributes` struct, where the `is_godzilla` flag is set to true, and the other fields are set to None or their default values. This approach effectively addresses the limitation where enum variants must share the same type.
```

Comment:
```suggestion
For the `Godzilla` variant, since it doesn't require any additional data, you would instantiate it with a `CharacterAttributes` struct, where the `is_godzilla` flag is set to true, and the other fields are set to `None` or their default values. This approach effectively addresses the limitation where enum variants must share the same type.
```